extensions/emmet/src/test/testUtils.ts

```diff
@@ -17,17 +17,11 @@ export function rndName() {
 	return name;
 }
 
-export function createRandomFile(contents = '', fileExtension = 'txt'): Thenable<vscode.Uri> {
-	return new Promise((resolve, reject) => {
-		const tmpFile = join(os.tmpdir(), rndName() + '.' + fileExtension);
-		fs.writeFile(tmpFile, contents, (error) => {
-			if (error) {
-				return reject(error);
-			}
-
-			resolve(vscode.Uri.file(tmpFile));
-		});
-	});
+export async function createRandomFile(contents = '', fileExtension = 'txt'): Promise<vscode.Uri> {
+	const tmpFile = join(os.tmpdir(), rndName() + '.' + fileExtension);
+	const fileUri = vscode.Uri.file(tmpFile);
+	await vscode.workspace.fs.writeFile(fileUri, Buffer.from(contents));
+	return fileUri;
```

Comment:
Using the FS API is always good but I doubt that will help to make tests more stable. 

For the API tests themselves we went away from tmp-file and OS dependencies. That reduces the dependencies of your tests. For the API tests stability was improved. 

This is how/where things are setup https://github.com/microsoft/vscode/blob/36fdf6b697cba431beb6e391b5a8c5f3606975a1/extensions/vscode-api-tests/src/utils.ts#L10-L17

Reply:
There is unfortunately no code reuse so you'll need to copy-paste this and the memfs-file